presto-spark-classloader-interface/src/main/java/com/facebook/presto/spark/classloader_interface/PrestoSparkTaskProcessor.java

```diff
@@ -0,0 +1,74 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark.classloader_interface;
+
+import org.apache.spark.TaskContext;
+import org.apache.spark.broadcast.Broadcast;
+import org.apache.spark.util.CollectionAccumulator;
+import scala.Tuple2;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.unmodifiableList;
+import static java.util.Collections.unmodifiableMap;
+import static java.util.Objects.requireNonNull;
+
+public class PrestoSparkTaskProcessor
+        implements Serializable
+{
+    private final PrestoSparkTaskExecutorFactoryProvider taskExecutorFactoryProvider;
+    private final SerializedPrestoSparkTaskDescriptor serializedTaskDescriptor;
+    private final List<String> fragmentIds;
+    private final CollectionAccumulator<SerializedTaskStats> taskStatsCollector;
+    // fragmentId -> Broadcast
+    private final Map<String, Broadcast<List<PrestoSparkSerializedPage>>> broadcastInputs;
+
+    public PrestoSparkTaskProcessor(
+            PrestoSparkTaskExecutorFactoryProvider taskExecutorFactoryProvider,
+            SerializedPrestoSparkTaskDescriptor serializedTaskDescriptor,
+            List<String> fragmentIds,
+            CollectionAccumulator<SerializedTaskStats> taskStatsCollector,
+            Map<String, Broadcast<List<PrestoSparkSerializedPage>>> broadcastInputs)
+    {
+        this.taskExecutorFactoryProvider = requireNonNull(taskExecutorFactoryProvider, "taskExecutorFactoryProvider is null");
+        this.serializedTaskDescriptor = requireNonNull(serializedTaskDescriptor, "serializedTaskDescriptor is null");
+        this.fragmentIds = unmodifiableList(new ArrayList<>(requireNonNull(fragmentIds, "fragmentIds is null")));
+        this.taskStatsCollector = requireNonNull(taskStatsCollector, "taskStatsCollector is null");
+        this.broadcastInputs = unmodifiableMap(new HashMap<>(requireNonNull(broadcastInputs, "broadcastInputs is null")));
+    }
+
+    public Iterator<Tuple2<Integer, PrestoSparkRow>> apply(
+            Iterator<SerializedTaskSource> serializedTaskSources,
+            List<Iterator<Tuple2<Integer, PrestoSparkRow>>> inputs)
+    {
+        int partitionId = TaskContext.get().partitionId();
+        int attemptNumber = TaskContext.get().attemptNumber();
+        Map<String, Iterator<Tuple2<Integer, PrestoSparkRow>>> inputsMap = new HashMap<>();
```

Comment:
nit: `shuffleInputsMap` or `fragmentIdToShuffleInput`?

Reply:
So essentially, these two information are maintained in different classes:
- `fragmentIds` list is maintained in `PrestoSparkTaskProcessor`
- `shuffleInput` list is maintained in `PrestoSparkTaskRdd`.

And these two lists are "zipped" when `PrestoSparkTaskRdd#compute` calls `PrestoSparkTaskProcessor#apply`. 

I am wondering if this "zipping" operation can be done in `PrestoSparkTaskRdd#compute`? -- Otherwise, it's a bit uneasy to reason the ordering of `shuffleInput` in `PrestoSparkTaskRdd` has to be consistent with the ordering of `fragmentIds` in `PrestoSparkTaskProcessor`... 

When I am reading the code, I have to traverse the code in the following way:
-  Oh, how to guarantee the ordering in this two list are consistent?
- ok , `shuffleInput` is from `PrestoSparkTaskRdd`, and it's from `PrestoSparkRddFactory`
- And  the same `PrestoSparkRddFactory` provides the fragment id list when construct `PrestoSparkTaskProcessor`
- OK so these two lists are consistent...

<img width="140" alt="Screen Shot 2020-06-08 at 11 14 12 PM" src="https://user-images.githubusercontent.com/799346/84112540-ccf9a200-a9dd-11ea-842c-d29ea6f1ebea.png">

Reply:
> I am wondering if this "zipping" operation can be done in PrestoSparkTaskRdd#compute? 

Could you please elaborate a little bit more. I believe I'm missing something. The list of `shuffleInput ` is only available as a parameter of `process` in `PrestoSparkTaskProcessor`. What zipping you think can be done in `PrestoSparkTaskRdd`?

Reply:
@arhimondr : 

> Could you please elaborate a little bit more. I believe I'm missing something. The list of shuffleInput is only available as a parameter of process in PrestoSparkTaskProcessor.

So essentially, you are zipping `fragmentIds` and `inputs` into `inputsMap`. And `fragmentIds` is a field in `PrestoSparkTaskProcessor`, while `inputs` is passed as a parameter into `PrestoSparkTaskProcessor#process`

Reply:
Yup

Reply:
@arhimondr : 

> Yup

But it's not fixed. I mean, the "zipped inputs" are from two different source (one stay in `PrestoSparkTaskProcessor` as a field member, one provided through `PrestoSparkTaskProcessor#process`). And it's quite confusing. For example, can they both be provided from `PrestoSparkTaskProcessor#process`? 

I would leave this as a future discussion for now.

Reply:
Nevermind, I think I got your point now. Refactored. Good catch.